test/e2e/performanceprofile/functests/13_llc/llc.go

```diff
@@ -102,58 +105,109 @@ var _ = Describe("[Performance] Last Level Cache", Label(string(label.OpenShift)
 			}
 		})
 
-		It("[test_id: 77723] Removing the kubelet annotation should disable align-cpus-by-uncorecache policy", func() {
-
+		AfterEach(func() {
+			testlog.Infof("Remove align-cpus-by-uncorecache cpumanager policy")
 			// Delete the Annotations
 			if perfProfile.Annotations != nil {
 				perfProfile.Annotations = nil
+
+				By("updating performance profile")
+				profiles.UpdateWithRetry(perfProfile)
+
+				By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+				profilesupdate.WaitForTuningUpdating(ctx, perfProfile)
+
+				By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
+				profilesupdate.WaitForTuningUpdated(ctx, perfProfile)
 			}
-			fmt.Println("AFter updating ", perfProfile.Annotations)
+		})
 
-			By("updating performance profile")
-			profiles.UpdateWithRetry(perfProfile)
+		When("align-cpus-by-uncorecache annotation is enabled", func() {
+			It("[test_id:77722] cpuManager Policy in kubelet is configured appropriately", func() {
+				for _, node := range workerRTNodes {
+					kubeletconfig, err := nodes.GetKubeletConfig(ctx, &node)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(kubeletconfig.CPUManagerPolicyOptions).To(HaveKeyWithValue("prefer-align-cpus-by-uncorecache", "true"))
+				}
+			})
+		})
 
-			By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
-			profilesupdate.WaitForTuningUpdating(ctx, perfProfile)
+		When("align-cpus-by-uncorecache annotation is removed", func() {
+			It("[test_id:77723] should disable align-cpus-by-uncorecache cpumanager policy", func() {
+				// Delete the Annotations
+				if perfProfile.Annotations != nil {
+					perfProfile.Annotations = nil
 
-			By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
-			profilesupdate.WaitForTuningUpdated(ctx, perfProfile)
+					By("updating performance profile")
+					profiles.UpdateWithRetry(perfProfile)
 
-			for _, node := range workerRTNodes {
-				kubeletconfig, err := nodes.GetKubeletConfig(ctx, &node)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(kubeletconfig.CPUManagerPolicyOptions).ToNot(HaveKey("prefer-align-cpus-by-uncorecache"))
-			}
+					By(fmt.Sprintf("Applying changes in performance profile and waiting until %s will start updating", poolName))
+					profilesupdate.WaitForTuningUpdating(ctx, perfProfile)
+
+					By(fmt.Sprintf("Waiting when %s finishes updates", poolName))
+					profilesupdate.WaitForTuningUpdated(ctx, perfProfile)
+				}
+
+				for _, node := range workerRTNodes {
+					kubeletconfig, err := nodes.GetKubeletConfig(ctx, &node)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(kubeletconfig.CPUManagerPolicyOptions).ToNot(HaveKey("prefer-align-cpus-by-uncorecache"))
+				}
+			})
 		})
 
-		It("[test_id: 77724] disabling align-cpus-by-uncorecache explicitly through annotation", func() {
+		When("align-cpus-by-uncorecache annotation is disabled", func() {
```

Comment:
Hmm, maybe we should say here that the policy option in disabled rather than annotation is disabled?

Reply:
fixed in the latest commit